test/client-utils.js

```diff
@@ -34,23 +34,24 @@ async function getTestClientForHs(testMatrixServerUrl) {
     }
   );
 
-  const userId = registerResponse['user_id'];
-  assert(userId);
+  const applicationServiceUserIdOverride = registerResponse['user_id'];
+  assert(applicationServiceUserIdOverride);
 
   return {
     homeserverUrl: testMatrixServerUrl,
     // We use the application service AS token because we need to be able to use
     // the `?ts` timestamp massaging when sending events
     accessToken: matrixAccessToken,
-    userId: userId,
+    userId: applicationServiceUserIdOverride,
+    applicationServiceUserIdOverride,
```

Comment:
Renamed to `applicationServiceUserIdOverride` so it's more obvious what this is used for, what user you want the application service to act on behalf of.

Also added `userId` to reference for the `updateProfile` Matrix API endpoints.

They are separate because we could have a API client which isn't driven by an application service.